content/en/blog/_posts/2023-10-24-taint-eviction-controller.md

```diff
@@ -0,0 +1,81 @@
+---
+layout: blog
+title: "Kubernetes 1.29: Decouple TaintManager from NodeLifecycleController"
+date: 2023-11-15
+slug: kubernetes-1-29-taint-eviction-controller
+---
+
+**Authors:** Yuan Chen (Apple), Andrea Tosatto (Apple)
+
+This blog discusses a new feature in Kubernetes 1.29 to improve the handling of taint-based pod eviction.
+
+## Background
+
+In Kubernetes 1.29, an improvement has been introduced to enhance the handling of taint-based pod eviction on nodes. This blog discusses the changes made to the `NodeLifecycleController` to separate its responsibilities and improve overall code maintainability.
+
+### Summary of Changes
+
+The `NodeLifecycleController` previously combined two independent functions:
+
+- Adding a pre-defined set of `NoExecute` taints to nodes based on the node conditions.
+- Performing pod eviction on NoExecute taints
+
+With the Kubernetes 1.29 release, the taint-based eviction implementation has been
+moved out of the `node-lifecycle-controller` into a separate and independent component called `TaintEvictionController`.
+This separation aims to disentangle code, enhance code maintainability,
+and facilitate future extensions to either component.
+
+As part of the change, additional metrics are introduced to monitor taint-based pod evictions.
+
+- `pod_deletion_duration_seconds` measures th latency between the time when a taint effect has been activated for the Pod and its deletion via `TaintEvictionController`.
```

Comment:
```suggestion
- `pod_deletion_duration_seconds` measures the latency between the time when a taint effect has been activated for a Pod and its deletion by the taint eviction controller.
```
Is this a counter metric? How do the labels work / where can people read more? (eg: https://kubernetes.io/docs/reference/instrumentation/metrics/)